enhancements/network/egress-router.md

```diff
@@ -0,0 +1,200 @@
+external services. The egress router acts as a bridge between pods and an
+external system.
+
+Traffic going out the egress router goes via node, but it will have the MAC
+address of the macvlan0 interface inside the egress router.
+
+In openshift-sdn, the egress router was implemented by adding an annotation to
+allow a pod to request a macvlan interface. In order to avoid repeating this
+behavior in ovn-kubernetes, we'd be requesting such interface using multus to
+ensure feature-parity with openshift-sdn.
+
+## Motivation
+
+* The egress router images allow people to run containers that have defined
+  external addresses.
+
+  Using network policies you can control the scope of who can talk to egress
+  routers and thus, when they send traffic outside the cluster it's possible
+  to identify the containers in the cluster which are sending it. This allows
+  to use external firewalls for auditing and filtering.
+
+  Using multus for requesting the macvlan interface also opens up the
+  possibility of use different interface types in the future.
+
+* Ensure that we have images that offer equivalent capabilities to the existing
+  egress router images when running ovn-kubernetes.
+
+### Goals
+
+- Create an egress router CNI plugin
+- Add feature parity with openshift-sdn's egress router.
+
+## Proposal
+
+## Design Details
+
+We'll be implenting this feature using an additional specific CNI plugin for
+it, whose [github repository](https://github.com/openshift/egress-router-cni/)
+is already created.
+
+Both kubernetes' pod and multus' net-attach-def are namespaced objects. Due to a
+security limitation in OCP, Pod yaml and net-attach-def object should be in the
+same namespace, thus egress-router-cni would net to take this into
+consideration and put them in the same namespace as well.
+
+The egress router pod would have multus annotations saying to use the
+egress-router-cni plugin, and then it would create a macvlan interface
+(or, in the future, maybe ipvlan or Amazon ENI).
+
+### Overview
+
+The `egress-router` plugin creates some sort of
+cluster-network-external network interface and assigns a user-provided
+static public IP address to it. It is designed for use by the
+OpenShift egress-router feature, but may have other uses as well.
+
+### Example configuration
+
+```
+{
+	"name": "egress-router-1",
+	"type": "egress-router",
+
+	"ip": {
+		"addresses": ["192.168.1.99/24"]
+	}
+}
+
+{
+	"name": "egress-router-2",
+	"type": "egress-router",
+
+	"interfaceType": "ipvlan",
+	"interfaceArgs": {
+		"master": "eth1",
+	},
+	"podIP": {
+		"db-router": {
+			"addresses": ["192.168.3.10/24"],
+			"gateway": "192.168.3.1",
+			"destinations": ["10.1.2.3/32"]
+		},
+		"metrics-router": {
+			"addresses": ["192.168.3.11/24"]
+			"gateway": "192.168.3.1",
+		},
+		"alpha-*": {
+			"addresses": ["192.168.3.12/24"]
+			"gateway": "192.168.3.1",
+		}
+	}
+}
+```
+
+### Network configuration reference
```

Comment:
We don't need this level of detail. You can explain the behavior of the above examples, and then refer to the docs in the repo for more details. Also, not all of the features documented here are even implemented.

Reply:
Yeah makes sense to defer this to the actual doc, a bit overkill for just the proposal, will just remove this whole section and add a link to the github CNI plugin repository.